helix-term/src/application.rs

```diff
@@ -927,10 +944,12 @@ impl Application {
                             .editor
                             .documents_mut()
                             .filter_map(|doc| {
-                                if doc.language_server().map(|server| server.id())
-                                    == Some(server_id)
+                                if doc
+                                    .language_servers()
+                                    .iter()
+                                    .any(|server| server.id() == server_id)
```

Comment:
I see this `any` pattern related a couple times; maybe it would be worth making an accessor function to factor it out? e.g. `fn language_server(id: &str) -> Option<_>`

Reply:
I think where possible, I have added a method `supports_language_server(&self, id: usize)`, the rest should be handled by the new `language_server_with_feature!` macro and generally filtering with `doc.language_servers_with_feature` now, so I guess this is resolved.